Doc/reference/datamodel.rst

```diff
@@ -2616,6 +2616,23 @@ are awaitable.
    :term:`awaitable` objects.  For instance, :class:`asyncio.Future` implements
    this method to be compatible with the :keyword:`await` expression.
 
+   .. note::
+
+      The language doesn't place any restrictions on the type or value of the
+      objects yielded by the iterator returned by `__await__`, as this is
+      specific to the implementation of the async framework that will be
+      managing the :term:`awaitable` object.
+      
+      In the case of :mod:`asyncio`, user code should generally be using
+      :term:`coroutines <coroutine>`, instances of :class:`asyncio.Task`,
+      :class:`asyncio.Future`, and other :mod:`asyncio` objects to implement
+      `__await__`; yielding objects from these (e.g.
+      `return (yield from future)`), rather than yielding objects directly.
+      An exception to this is a bare `yield` within an `__await__`, which can
+      be used to suspend execution until the next iteration of the event loop.
```

Comment:
Also mention explicitly yielding `None`, and mention the bare `yield` as an equivalent form

```suggestion
      An exception to this is a to yield `None`—or the equivalent bare `yield`—, within an `__await__`,
      which can be used to suspend execution until the next iteration of the
      event loop.
```

Reply:
I've replaced a long dash symbol with brackets, but the sentence still look a little long and awkward IMHO

Reply:
@asvetlov I think the latest version with the added semicolon and splitting it into two sentences looks reasonably decent. 

@plammens 
> Also mention explicitly yielding None, and mention the bare yield as an equivalent form

Yep, that makes sense to me. In your offered suggestion, I would recommend changing it slightly to the following (fixes minor grammar issue and line width):

```suggestion
      An exception to this is a `yield None` (or equivalent bare `yield`)
      within an `__await__`, which can be used to suspend execution until the
      next iteration of the event loop.
```